### PR TITLE
fix(tests): Update AMIs

### PR DESCRIPTION
**Pull request by amolero-nr**
Update several test definition files with a newer AMI name.

### CODE REVIEW

File: test/definitions-eu/otel/rhel/awslinux23-otel.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "resources": [{
         "id": "host1",
         "provider": "aws",
-        "ami_name": "al2023-ami-2023.0.*-x86_64",
+        "ami_name": "al2023-ami-2023.1.*-x86_64",
```
Comment (rthorn-nr) [inline test/definitions-eu/otel/rhel/awslinux23-otel.json:12]: Is this digit something we'd want to wildcard? Would be a pain if this changes to `2`

Comment (amolero-nr) [reply test/definitions-eu/otel/rhel/awslinux23-otel.json:12]: I guess we could, but maybe is useful to make sure to test with the oldest minor version of the AL2023 AMI?

From what I [read](https://docs.aws.amazon.com/linux/al2023/ug/naming-and-versioning.html), when version `2` is released (e.g.: `al2023-ami-2023.2.*-x86_64`), version `1` is not automatically removed, they will be there for some time (standard and extended support).

Comment (rthorn-nr) [reply test/definitions-eu/otel/rhel/awslinux23-otel.json:12]: That sounds reasonable to me, even though it will mean another update in the future

**Review by rthorn-nr — DISMISSED** [review]

File: test/definitions-eu/otel/rhel/awslinux23-otel.json (hunk 1 above)
Comment (amolero-nr) [reply test/definitions-eu/otel/rhel/awslinux23-otel.json]: Done in 9c1ca48f8772aace21f16d4b7a4def96a8f42107

**Review by rthorn-nr — APPROVED** [review]

File: test/definitions-eu/otel/rhel/awslinux23-otel.json [hunk 2]
```diff
@@ -7,9 +7,9 @@
     },
 
     "resources": [{
-        "id": "host1",
+        "id": "host2",
```
Comment (NSSPKrishna) [inline test/definitions-eu/otel/rhel/awslinux23-otel.json:10]: 👋 , is there a specific reason behind changing the id?

**Pull request merged**